crates/adapters/src/catalog.rs

```diff
@@ -421,31 +426,61 @@ impl Catalog {
         }
     }
 
-    pub fn register_input_collection_handle<H>(&mut self, name: &str, handle: H)
-    where
-        H: DeCollectionHandle + 'static,
-    {
-        self.input_collection_handles
-            .insert(name.to_owned(), Box::new(handle));
+    pub fn register_input_collection_handle(&mut self, name: &str, handle: InputCollectionHandle) {
+        let name = if handle.schema.case_sensitive {
+            name.to_string()
+        } else {
+            name.to_lowercase()
+        };
+
+        self.input_collection_handles.insert(name, handle);
```

Comment:
can this fail?
does this overwrite a previously existing name?

Reply:
yes, there's no checking for duplicates at the moment. I'll add that.